Unit test case 'executor TestIndexMergeRuntimeStats' fail on real TiKV

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

I'm [testing paging protocol](https://github.com/pingcap/tidb/issues/35633) by running tidb UT with a real TiKV using this branch https://github.com/pingcap/tidb/pull/35647



```
tiup playground nightly --mode tikv-slim
cd executor
go test -run TestIndexMergeRuntimeStats -with-tikv 127.0.0.1:2379
```

### 2. What did you expect to see? (Required)

No error

### 3. What did you see instead (Required)

```
--- FAIL: TestIndexMergeRuntimeStats (3.79s)
    testkit.go:174: 
        	Error Trace:	testkit.go:174
        	            				testkit.go:168
        	            				testkit.go:108
        	            				executor_test.go:3070
        	Error:      	Received unexpected error:
        	            	runtime error: invalid memory address or nil pointer dereference
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask.func1
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:664
        	            	runtime.gopanic
        	            		/home/genius/project/go/src/runtime/panic.go:838
        	            	runtime.panicmem
        	            		/home/genius/project/go/src/runtime/panic.go:220
        	            	runtime.sigpanic
        	            		/home/genius/project/go/src/runtime/signal_unix.go:818
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleCopResponse
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:961
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTaskOnce
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:786
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:674
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).run
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:416
        	            	runtime.goexit
        	            		/home/genius/project/go/src/runtime/asm_amd64.s:1571
        	Test:       	TestIndexMergeRuntimeStats
        	Messages:   	sql:select /*+ use_index_merge(t1, primary, t1a) */ * from t1 where id < 2 or a > 4 order by a, args:[]
FAIL
```

I've verified it's not caused by the changes from paging... instead, it's a bug on execution_info with cop cache.

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

master 

```
commit 47a4f3a017768dda0b9a826451486c5539bc8427 (HEAD -> master, origin/master, origin/HEAD)
Author: Weizhen Wang <wangweizhen@pingcap.com>
Date:   Wed Jun 29 20:16:40 2022 +0800

    *: enable unconvert (#35821)
    
    ref pingcap/tidb#35345
```

--- a/store/copr/coprocessor.go
+++ b/store/copr/coprocessor.go
@@ -958,17 +958,19 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *tikv.R
 	} else {
 		// Cache not hit or cache hit but not valid: update the cache if the response can be cached.
 		if cacheKey != nil && resp.pbResp.CanBeCached && resp.pbResp.CacheLastVersion > 0 {
-			if worker.store.coprCache.CheckResponseAdmission(resp.pbResp.Data.Size(), resp.detail.TimeDetail.ProcessTime) {
-				data := make([]byte, len(resp.pbResp.Data))
-				copy(data, resp.pbResp.Data)
+			if resp.detail != nil {
+				if worker.store.coprCache.CheckResponseAdmission(resp.pbResp.Data.Size(), resp.detail.TimeDetail.ProcessTime) {
+					data := make([]byte, len(resp.pbResp.Data))
+					copy(data, resp.pbResp.Data)
 
-				newCacheValue := coprCacheValue{
-					Data:              data,
-					TimeStamp:         worker.req.StartTs,
-					RegionID:          task.region.GetID(),
-					RegionDataVersion: resp.pbResp.CacheLastVersion,
+					newCacheValue := coprCacheValue{
+						Data:              data,
+						TimeStamp:         worker.req.StartTs,
+						RegionID:          task.region.GetID(),
+						RegionDataVersion: resp.pbResp.CacheLastVersion,
+					}
+					worker.store.coprCache.Set(cacheKey, &newCacheValue)
 				}
-				worker.store.coprCache.Set(cacheKey, &newCacheValue)
 			}
 		}
 	}

--- a/tests/realtikvtest/sessiontest/session_test.go
+++ b/tests/realtikvtest/sessiontest/session_test.go
@@ -3728,3 +3728,29 @@ func TestBinaryReadOnly(t *testing.T) {
 	require.Equal(t, 2, session.GetHistory(tk.Session()).Count())
 	tk.MustExec("commit")
 }
+
+func TestIndexMergeRuntimeStats(t *testing.T) {
+	store, clean := realtikvtest.CreateMockStoreAndSetup(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("set @@tidb_enable_index_merge = 1")
+	tk.MustExec("create table t1(id int primary key, a int, b int, c int, d int)")
+	tk.MustExec("create index t1a on t1(a)")
+	tk.MustExec("create index t1b on t1(b)")
+	tk.MustExec("insert into t1 values(1,1,1,1,1),(2,2,2,2,2),(3,3,3,3,3),(4,4,4,4,4),(5,5,5,5,5)")
+	rows := tk.MustQuery("explain analyze select /*+ use_index_merge(t1, primary, t1a) */ * from t1 where id < 2 or a > 4;").Rows()
+	require.Len(t, rows, 4)
+	explain := fmt.Sprintf("%v", rows[0])
+	pattern := ".*time:.*loops:.*index_task:{fetch_handle:.*, merge:.*}.*table_task:{num.*concurrency.*fetch_row.*wait_time.*}.*"
+	require.Regexp(t, pattern, explain)
+	tableRangeExplain := fmt.Sprintf("%v", rows[1])
+	indexExplain := fmt.Sprintf("%v", rows[2])
+	tableExplain := fmt.Sprintf("%v", rows[3])
+	require.Regexp(t, ".*time:.*loops:.*cop_task:.*", tableRangeExplain)
+	require.Regexp(t, ".*time:.*loops:.*cop_task:.*", indexExplain)
+	require.Regexp(t, ".*time:.*loops:.*cop_task:.*", tableExplain)
+	tk.MustExec("set @@tidb_enable_collect_execution_info=0;")
+	tk.MustQuery("select /*+ use_index_merge(t1, primary, t1a) */ * from t1 where id < 2 or a > 4 order by a").Check(testkit.Rows("1 1 1 1 1", "5 5 5 5 5"))
+}

